lib/hyrax/transactions/steps/save.rb

```diff
@@ -53,10 +56,54 @@ def call(change_set, user: nil)
           publish_changes(resource: saved, user: user, new: unsaved.new_record, new_collections: new_collections)
           Success(saved)
         end
-        # rubocop:enable Metrics/MethodLength
+        # rubocop:enable Metrics/MethodLength, Metrics/AbcSize
 
         private
 
+        def valid_future_date?(item, attribute)
+          raise StandardError, "#{item.model} must use a future date" if item.fields[attribute] < Time.zone.now
+        end
+
+        def save_leases_and_embargoes(unsaved)
+          if unsaved.embargo.present?
+            unsaved.embargo.embargo_release_date = unsaved.embargo.embargo_release_date&.to_datetime
+            unsaved.embargo = @persister.save(resource: unsaved.embargo)
+          end
+          return if unsaved.lease.blank?
+
+          unsaved.lease.lease_expiration_date = unsaved.lease.lease_expiration_date&.to_datetime
+          unsaved.lease = @persister.save(resource: unsaved.lease)
+        end
+
+        # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+        def add_lease_to_members(change_set)
+          # TODO: account for all members and levels, ref: #6131
+          # use recursion to ensure the change_set that applies to the member is getting saved and published/indexed
+          change_set.model.member_ids&.each do |member|
+            item = Hyrax.query_service.find_by(id: member.id)
+            next unless item.is_a? Hyrax::FileSet
+
+            lease_updates = change_set.lease.instance_variable_get(:@_changes).values.select { |v| v }.any?
```

Comment:
``` bash
# example of what change_set.lease.instance_variable_get(:@_changes) could look like
@_changes={"visibility_after_lease"=>false, "visibility_during_lease"=>false, "lease_expiration_date"=>true}
```

Reply:
I moved this method into the LeaseManager. ref: https://github.com/samvera/hyrax/pull/6127/commits/b2a9f2303dd67e4f0a4b560da82f84d574fd84fd